lib/schedule/health/server.ex

```diff
@@ -1,11 +1,11 @@
-defmodule Schedule.HealthServer do
+defmodule Schedule.Health.Server do
   @moduledoc """
   GenServer to keep track of whether the GTFS GenServer is up and running
   """
 
   use GenServer
 
-  @type state :: :not_loaded | :loaded
+  @type state :: :not_loaded | :loaded | :unhealthy | :healthy
```

Comment:
It'd be possible to do this with only two states by checking it on the `:loaded` cast instead of the `:ready?` call:

```ex
@type state :: :ready | :not_ready

def handle_cast(:loaded) do
  healthy? = checker_healthy_fn.()
  new_state = if healthy?, do: :ready, else: :not_ready
  {:noreply, new_state}
end